### ISSUE TITLE
json-ld breadcrumb console error

### ISSUE THREAD

**Issue by zugrina** [sybrew/the-seo-framework#347]
Hello,

We have a breadcrumb json-ld error because of missing @id for the current item:

https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fwww.kaspersky.com%2Fblog%2Fmobile-endpoint-security%2F23479%2F

We saw your comment about @id:

> Another very important change is regarding the breadcrumb generation. WooCommerce 3.0 added new LD-JSON scripts at the bottom of the page. Because TSF outputs the scripts with an identifier related to the same page, the ones with product specifications from WooCommerce were seen as unbound.

> This issue resulted in that Google notoriously ignored WooCommerce’s scripts. Another effect, because multiple breadcrumb scripts were outputted, was that the scripts gave off incorrect signals to Google and other search engines. This was because the scripts were erratically referring.

> With 3.0, these issues should no longer arise. In short: What I’ve done is limiting the breadcrumb scripts to one from many, and I’ve removed the current page’s identifier.

Should we be concerned because of that or that error can be ignored?

Also, we did not find a hook how to return a missing @id, any idea how we can do that?

Thank you!

**Comment by sybrew** [comment]
Thank you for opening this issue!

_tl;dr: This is a recent addition in the testing tool. It's not yet implemented in the search engine._

------------------
**Before we continue, I do want to highlight this:**

Google's Search Console will assess the "real" validity of the items. 

The testing tool seems to be inconsistent, and sometimes even incorrect ([Articles](https://theseoframework.com/extensions/articles/#changelog/1-0-1-gamma)) in its assessments.

With this, we can confirm that the testing tool's implementation is not used for real-world applications. And, **you can safely ignore this issue** until further notice.

------------------

Now, this seems to be a recently planned requirement for the `itemListElement`. Google's [expanding on this type for other implementations](https://developers.google.com/search/docs/guides/mark-up-listings).
Microdata-powered breadcrumbs (like outputted by the Genesis theme) are also suddenly marked invalid.

I'm going to investigate this issue; so to confirm whether this will be used by Google in the near future. I'm afraid I'll have to reintroduce the `@id` for the latest breadcrumb, and I'll have to retest the conflict with WooCommerce.

**Comment by sybrew** [comment]
WooCommerce added breadcrumbs to their own output; I don't know when this happened.

I could either:
1. Disable their breadcrumbs when TSF is active. Because TSF's breadcrumbs supersede theirs (but we might run into the same conflict, _again_).
2. Disable TSF's breadcrumbs on WC product page.

We'll definitely go with 2. if the conflict is found.

As an aside, their plugin also suffers from Google's new addition. And I've forwarded this issue to them:
https://github.com/woocommerce/woocommerce/issues/21152

**Comment by zugrina** [comment]
Thank you @sybrew!
